### PR TITLE
Ft user specialist mapping

### LINKED ISSUE #5: [FEATURE] Add user-specialist mapping
### Add the following user-specialist mapping related tasks:
1. Specialization model, validation, and tests.
2. Specialization controller and tests.
3. Specialization Mapper model, validation, and tests.
4. Specialization Mapper controller and tests.

### PR DESCRIPTION
**Pull request by wathigo**
## Description
1. Added SpecialistMapper model, associations, and validations.
2. Added SpecialistMapper controller.
3. Added unit and integration tests.

Fixes #5 

## How Has This Been Tested?
1. Clone the repository.
2. Check out to `ft-user-specialist-mappings` branch.
3. Run `bundle install`.
4. To set up the database, run `rails db: setup`
4. Run `rails db:migrate`  to create the database schema.
5. Enter `bundle exec rspec` to run the tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes
****

### CODE REVIEW

File: app/controllers/specialist_mappers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class SpecialistMappersController < ApplicationController
+  before_action :set_specialist_mapper, only: %i[show update destroy]
+
+  def index
+    json_response(SpecialistMapper.all)
```
Comment (Danielshow) [inline app/controllers/specialist_mappers_controller.rb]: What if the Specialist is much, can we paginate the data with will_paginate or another gem

**Review by Danielshow — CHANGES_REQUESTED** [review]
Thanks for the linting. Can you work on the comments on paginating

File: app/controllers/specialist_mappers_controller.rb (hunk 1 above)
Comment (wathigo) [reply app/controllers/specialist_mappers_controller.rb]: Sure, Let me work on that.

**Review by katungi — DISMISSED** [review]
LGTM!

**Comment by wathigo** [comment]
@Danielshow Requested changes have been implemented.

**Review by Danielshow — APPROVED** [review]
Awesome thank you

**Review by katungi — APPROVED** [review]
amazing. LGTM!!

**Pull request merged**